google/cloud/testing_util/validate_metadata.cc

```diff
@@ -333,11 +334,14 @@ absl::optional<std::string> ValidateMetadataFixture::GetAuthority(
 }
 
 void ValidateMetadataFixture::SetServerMetadata(
-    grpc::ClientContext& client_context, RpcMetadata const& server_metadata) {
+    grpc::ClientContext& client_context, RpcMetadata const& server_metadata,
+    bool is_initial_metadata_ready) {
   // Create a server context with the given server metadata.
   grpc::GenericServerContext server_context;
-  for (auto const& kv : server_metadata.headers) {
-    server_context.AddInitialMetadata(kv.first, kv.second);
+  if (is_initial_metadata_ready) {
```

Comment:
This thing is still exchanging the metadata. `grpc::ClientContext::initial_metadata_received_` will still be `true`.

All you are saying is to skip adding the metadata, which the caller could decide to do by leaving `server_metadata.headers` empty.

Reply:
I removed these changes, so I think this is no longer relevant, but lmk if I am wrong